pkg/operator/ceph/spec/spec.go

```diff
@@ -145,3 +146,34 @@ func PodLabels(appName, namespace, daemonType, daemonID string) map[string]strin
 	labels[daemonType] = daemonID
 	return labels
 }
+
+// CheckPodMemory verify pod's memory limit is valid
+func CheckPodMemory(resources v1.ResourceRequirements, cephPodMinimumMemory uint64) error {
+	// Ceph related PR: https://github.com/ceph/ceph/pull/26856
+	podMemoryLimit := resources.Limits.Memory()
+	podMemoryRequest := resources.Requests.Memory()
+	errorMessage := `refuse to run the pod with %dmb of ram, provide at least %dmb.`
+
+	// If nothing was provided let's just return
+	// This means no restrictions on pod's resources
+	if podMemoryLimit.IsZero() && podMemoryRequest.IsZero() {
+		return nil
+	}
+
+	// This means LIMIT and REQUEST are either identical or different but still we use LIMIT as a reference
+	if uint64(podMemoryLimit.Value()) < display.MbTob(cephPodMinimumMemory) {
+		return fmt.Errorf(errorMessage, display.BToMb(uint64(podMemoryLimit.Value())), cephPodMinimumMemory)
+	}
+
+	// This means LIMIT < REQUEST
+	// Kubernetes will refuse to schedule that pod however it's still valuable to indicate that user's input was incorrect
+	if uint64(podMemoryLimit.Value()) < display.MbTob(cephPodMinimumMemory) {
```

Comment:
@leseb is this a typo? This condition is exactly the same as the previous one.

Reply:
Fixed in https://github.com/rook/rook/pull/3258, thanks!